gcc/rust/lex/rust-token.cc

```diff
@@ -172,6 +173,41 @@ Token::get_str () const
   return *str;
 }
 
+namespace {
+enum class Context
+{
+  String,
+  Char
+};
+
+const std::map<char, std::string> matches = {
+  {'\t', "\\t"}, {'\n', "\\n"},	 {'\r', "\\r"},
+  {'\0', "\\0"}, {'\\', "\\\\"}, {'\v', "\\v"},
+};
+
+std::string
+escape_special_chars (const std::string &source, Context ctx)
+{
+  std::stringstream stream;
+  decltype (matches)::const_iterator result;
+  for (char c : source)
+    {
+      // FIXME: Also replace escaped unicode values and \x digits
```

Comment:
Put in the issue number? last nit and I'll merge this haha

Reply:
Done